src/test/java/org/breedinginsight/brapi/v2/BrAPIV2ControllerIntegrationTest.java

```diff
@@ -0,0 +1,391 @@
+
+    @Inject
+    private DSLContext dsl;
+    @Inject
+    private ProgramDao programDao;
+    @Inject
+    private UserDAO userDAO;
+
+    private ProgramEntity validProgram;
+
+    @BeforeAll
+    @SneakyThrows
+    public void setup() {
+        fp = FannyPack.fill("src/test/resources/sql/BrapiObservationVariablesControllerIntegrationTest.sql");
+        var securityFp = FannyPack.fill("src/test/resources/sql/ProgramSecuredAnnotationRuleIntegrationTest.sql");
+
+        // Insert system roles
+        User testUser = userDAO.getUserByOrcId(TestTokenValidator.TEST_USER_ORCID)
+                               .get();
+        dsl.execute(securityFp.get("InsertSystemRoleAdmin"),
+                    testUser.getId()
+                            .toString());
+
+        // Insert program
+        dsl.execute(fp.get("InsertProgram"));
+
+        // Insert program observation level
+        dsl.execute(fp.get("InsertProgramObservationLevel"));
+
+        // Insert program ontology sql
+        dsl.execute(fp.get("InsertProgramOntology"));
+        dsl.execute(fp.get("InsertTestProgramUser"));
+        dsl.execute(fp.get("InsertOtherTestProgramUser"));
+
+        // Retrieve our new data
+        validProgram = programDao.findAll()
+                                 .stream()
+                                 .filter(programEntity -> programEntity.getName()
+                                                                       .equals("Test Program"))
+                                 .findFirst()
+                                 .get();
+
+        dsl.execute(securityFp.get("InsertProgramRolesBreeder"),
+                    testUser.getId()
+                            .toString(),
+                    validProgram.getId()
+                                .toString());
+
+        var brapiFp = FannyPack.fill("src/test/resources/sql/brapi/species.sql");
+        super.getBrapiDsl()
+             .execute(brapiFp.get("InsertSpecies"));
+    }
+
+    @Test
+    public void testRootServerInfo() {
+        Flowable<HttpResponse<String>> call = biClient.exchange(GET("/brapi/v2/serverinfo"), String.class);
+
+        HttpResponse<String> response = call.blockingFirst();
+        assertEquals(HttpStatus.OK, response.getStatus());
+        assertNotNull(response.body(), "Response body is empty");
+
+        JsonObject result = JsonParser.parseString(response.body())
+                                      .getAsJsonObject()
+                                      .getAsJsonObject("result");
+        BrAPIServerInfo serverInfo = GSON.fromJson(result, BrAPIServerInfo.class);
+
+        assertEquals("Breeding Insight Platform", serverInfo.getOrganizationName());
+        assertEquals("bi-api", serverInfo.getServerName());
+        assertEquals("bidevteam@cornell.edu", serverInfo.getContactEmail());
+        assertEquals("breedinginsight.org", serverInfo.getOrganizationURL());
+    }
+
+    @Test
+    public void testPostGetVariablesProxy() throws Exception {
```

Comment:
Think you could also use @SneakyThrows if you want

Reply:
Changed and pushed